### ISSUE TITLE
DLL sidedeck .x files with Swift 4.0.1 vs 4.2.1

### ISSUE THREAD

**Issue by travatine** [IBM-Swift/IBM-Swift-zOS#4]
Hi,

I have a Swift Package Manager package that is built on z/OS.
When I built the package using Swift 4.0.1 a DLL and associated ".x" file was output by the "swift build" .
The ".x" file contained text, specifically the functions exported by the Swift DLL.

When I build the same package with Swift 4.2.1 with no other changes ( other than the compiler version), the DLL is created and appears to be ok, however the ".x" sidedeck file is no longer generated as text ( listing the exported functions), instead it appears to be a program object / load module ; or at least the functions exported by the DLL do not appear in the file as text.

Are new command line options required with Swift 4.2.1 to generate DLLs using the Swift package manager?

Thanks again,

Kind Regards,
Travis Thorne

**Comment by travatine** [comment]
The Swift "Package.swift" file:

```
// swift-tools-version:4.0
// The swift-tools-version declares the minimum version of Swift required to build this package.

import PackageDescription

let package = Package(
    name: "MyApi",
    products: [
        // Products define the executables and libraries produced by a package, and make them visible to other packages.
        .library(
            name: "MyApi",
            type: .dynamic,
            targets: ["MyApi"]),
    ],
    dependencies: [
        // Dependencies declare other packages that this package depends on.
        // .package(url: /* package url */, from: "1.0.0"),
        .package(url: "../MyApiNative", .branch("master"))
    ],
    targets: [
        // Targets are the basic building blocks of a package. A target can define a module or a test suite.
        // Targets can depend on other targets in this package, and on products in packages which this package depends on.
        .target(
            name: "MyApi",
            dependencies: ["MyApiNative"]
            ),
        .testTarget(
            name: "MytApiTests",
            dependencies: ["MyApi"]),
    ]
)
```

The command I use to build the package:
```
swift build -v -c release \
    -Xswiftc -emit-library \
    -Xswiftc -emit-module \
    -Xlinker "`pwd`/../MyApiNative/.build/s390x-ibm-zos/release/libMyApiNative.a" 
```

libMyApiNative.a is a C based swift Package (static library) that builds successfully using both Swift 4.0.1 and Swift 4.2.1

**Comment by IgorTodorovskiIBM** [comment]
We released a new update for 4.2.1 in December which includes a fix to the swift-post-link tool which should address this problem.  Can you confirm your build with:
`swiftc -v`
```
5655-SFT V4.2 - IBM Toolkit for Swift on z/OS version 4.2.1-dev (LLVM 293f7fd05d, Clang bf22bdb497, Swift 66aef69d6c)
Build Date: Dec 12 2018 11:10:23
Target: s390x-ibm-zos
```

**Comment by travatine** [comment]
Hi,

Yes, it appears we've got the swift-v4.2.1-201811252016  version.
I've asked our sysprogs to install the Dec 12 update.

Thanks again,
Travis
